File: frontend/services/APIModel/Project/CreateProjectVersioningModel.ts
import NugetPackageModel from "../Shared/NugetPackageModel";

export default interface CreateProjectVersioningModel {
    projectName: string,
    projectWebsiteUrl: string,
    urlPrefix: string,
    gitDocsRepoUrl: string,
    gitDocsBranchName: string,
    gitDocsRelativePath: string,
    gitHomepageRelativePath: string,
    autoupgrage: boolean,
    nugetPackages: NugetPackageModel[] | null
}

File: backend/DNDocs.Web/Models/Project/CreateProjectVersioningModel.cs
using DNDocs.Web.Resources;
using DNDocs.API.Model.Shared;
using System.ComponentModel.DataAnnotations;

namespace DNDocs.Web.Models.Project
{
    public class CreateProjectVersioningModel : DNDocs.API.Model.Project.CreateProjectVersioningModel
    {
        [Required(ErrorMessage = "FormFieldRequired")]
        public override string ProjectName { get; set; }
        
        [Required(ErrorMessage = "FormFieldRequired")]
        public override string ProjectWebsiteUrl { get; set; }

        public override string GitDocsRepoUrl { get; set; }
        public override string GitDocsBranchName { get; set; }
        public override string GitDocsRelativePath { get; set; }
        public override string GitHomepageRelativePath { get; set; }
        public override IList<NugetPackageModel> NugetPackages { get; set; }
        public override bool Autoupgrage { get; set; }

        public CreateProjectVersioningModel() { }
    }
}
